common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -316,6 +388,19 @@ static ByteBuffer allocateDirectNoCleaner(int capacity) {
         return newDirectBuffer(UNSAFE.allocateMemory(capacity), capacity);
     }
 
+    static byte[] allocateUninitializedArray(int size) {
+        if (ALLOCATE_ARRAY_METHOD == null) {
+            return new byte[size];
+        }
+        try {
+            return (byte[]) ALLOCATE_ARRAY_METHOD.invoke(INTERNAL_UNSAFE, byte.class, size);
```

Comment:
Do you have benchmarks to show the impact of using reflection here? I'm concerned that we're trading the cost of memset for the cost of a reflective call.

Ideally, we would just call `Unsafe. allocateUninitializedArray` but that mean that we would have to require Java 9 to build.